### PR TITLE
[Dbx 88811] add paranoid success polling for 526

### PR DESCRIPTION
**Pull request by SamStuckey**
# 2nd attempt
The first was merged and reverted. this includes a syntax fix for the Sidekick job config. The last version of this attempted to use this syntax incorrectly. I've made two changes here, first is to make `args` into `'args'` which fixes the intial no method error. The second is to forgo using named arguments so my syntax is a 1 to 1 match with the docs.

I've ran through this in a staging command line to confirm there are no syntax errors. However, there is no way (that I know of) to make sure this job will actually receive it's arguments correctly using this syntax until it actually runs. Thankfully, that is perfectly safe. If the job fails on Sunday, we will have a nice error message in Datadog telling us what went wrong and the application will be unaffected.

cc @rmtolmach who caught the error the first time around

### EDIT: named args
Code Climate doesn't like that i've removed the named parameters. reverting to that paradigm. This is still safe

## Summary

Add a new `backup_submitted_claim_status` value "paranoid_success" to represent Benefits Intake's 'success' status
which requires special consideration. We need to consider it 'success type', as it will never progress to 'vbms', which is our true BI success state. It can however later revert to a failure state in BI, and so we need to continue polling on it for up to 1 year.

- *This work is behind a feature toggle (flipper): NO*
- We will now treat a Benefits Intake status of `success` as true success in our system
- This unblocks "definition of success" work that is foundational to 526 state monitoring
- DBX team carbs

Additional:
- make `last_remediation` method public on 526, this is useful for the manual remediation scripts

## Related issue(s)

- [Ticket](https://app.zenhub.com/workspaces/disability-benefits-experience-team-carbs-6470c8bfffee9809b2634a52/issues/gh/department-of-veterans-affairs/va.gov-team/88811)
- [Document explaining what this is and why we need it](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/engineering_research/paranoid_success_submissions.md)
- [Document on 526 state and reporting](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/engineering_research/untouched_submission_audit/526_state_repair_tdd.md)

## Testing done

- [x] *New code is covered by unit tests*
- Manually tested in production command line as part of remediation backfil

## What areas of the site does it impact?
form 526 state

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (comming soon)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature (n/a)

### CODE REVIEW

**Review by tblackwe — APPROVED** [review]
🎵 Everybody's coming to get Sam. 🎵 

Approved.

**Review by freeheeling — APPROVED** [review]
lgtm

File: lib/periodic_jobs.rb
Comment (rmtolmach) [inline lib/periodic_jobs.rb:47]: @SamStuckey A few things:
1. Why don't we have specs for this file, as described in [the docs](https://github.com/sidekiq/sidekiq/wiki/Ent-Periodic-Jobs#testing)? 😕 I'm not saying write all the specs for this file, but did you try to write up a little test for this?
2. Is `=>` right? I don't see that syntax in the docs (linked above). I see the `:` syntax. Oh wait. Maybe [I'm wrong](https://github.com/sidekiq/sidekiq/wiki/Ent-Periodic-Jobs#notes)
3. why in the world do we comment _below_ each line in this file?? 😆 😂

Comment (SamStuckey) [reply lib/periodic_jobs.rb:47]: 1. good call I'll add a spec
2. `:` is my preference, but hash rocket is what they have in the docs and should do the same thing so ¯\_(ツ)_/¯ 
3. yeah that annoyed me too... i thought about moving them all but that would've made a much bigger PR

Comment (SamStuckey) [reply lib/periodic_jobs.rb:47]: Ok, figured  out why there's no spec 😓 it's because configuring it is weird... i'm having trouble getting the sidekiq jobs to be available in the spec file. the  spec i tried looks like this

```
require 'rails_helper'

Dir[Rails.root.join('app/sidekiq/**/*')].sort.each { |file| require file }

RSpec.describe 'Periodic Jobs', type: :request do
  it 'verifies that periodic jobs have been defined with correct syntax' do
    require 'periodic_jobs'
    require 'sidekiq-ent/periodic/testing'
    ct = Sidekiq::Periodic::ConfigTester.new
    ct.verify(&PERIODIC_JOBS)
  end
end
```

but i'm getting  'cannot load ...' type errors.

Comment (SamStuckey) [reply lib/periodic_jobs.rb:47]: Can we ticket this for later work? feels out of scope to me, but I defer to you

Comment (rmtolmach) [reply lib/periodic_jobs.rb:47]: totally.

**Review by rmtolmach — APPROVED** [review]
Looks good! After merge, watch to see if your commit passes its tests again: https://github.com/department-of-veterans-affairs/vets-api/commits/master/ then when that's done, (wait a few min and then) check Argo dev to make sure it syncs and everything is 💚 https://argocd.vfs.va.gov/applications/vets-api-dev?resource=

**Pull request merged**
